pymc3/tests/test_sampling.py

```diff
@@ -971,14 +968,14 @@ def test_multivariate2(self):
 
     def test_layers(self):
         with pm.Model() as model:
-            a = pm.Uniform("a", lower=0, upper=1, size=10)
-            b = pm.Binomial("b", n=1, p=a, size=10)
+            a = pm.Uniform("a", lower=0, upper=1, size=5)
+            b = pm.Binomial("b", n=1, p=a, size=7)
 
         model.default_rng.get_value(borrow=True).seed(232093)
 
         b_sampler = aesara.function([], b)
         avg = np.stack([b_sampler() for i in range(10000)]).mean(0)
-        npt.assert_array_almost_equal(avg, 0.5 * np.ones((10,)), decimal=2)
+        npt.assert_array_almost_equal(avg, 0.5 * np.ones((7, 5)), decimal=2)
```

Comment:
Again, why were these changed?

Reply:
This is another example where the previous implementation did not actually treat `size` as being _in addition to_ implied dimensions:
```
a = Uniform(..., size=5)         # this is now a vector
b = Binomial(..., p=a, size=7)   # ndim=1 is already implied through `p`; size scales it up to a matrix!
```
Using non-equal sizes like the prime numbers 5 and 7 makes the test easier to debug in case some dimensions are flipped.